src/EditorFeatures/Core.Wpf/Suggestions/SuggestedActionsSource.cs

```diff
@@ -169,7 +168,9 @@ public bool TryGetTelemetryId(out Guid telemetryId)
                     //       IWaitIndicator abstraction (which is a thin wrapper on top of IVsThreadedWaitDialog) directly
                     //       for now, we use the one LB created before calling us. meaning we don't update
                     //       text on the wait dialog window
-                    _workspaceStatusService.WaitUntilFullyLoadedAsync(cancellationToken).Wait(cancellationToken);
+                    // 
+                    //       this also needs to run under threading context otherwise, we can deadlock on VS
```

Comment:
Any idea how we can ensure people *do* use JTF with this one as opposed to assuming Roslyn style waiting?

Reply:
I am not sure. I think at some point, we might always need to use _threadingContext for wait and based on host, it might or might not use JTF for waiting...